### PR TITLE
wgengine/monitor: close winMin.closeHandle

### PR DESCRIPTION
**Pull request by alexbrainman**
eccc167 introduced winMin.closeHandle. The commit opens the handle,
but it never closes it.

Windows handles are not boundless, and should be closed.

Updates #921

Signed-off-by: Alex Brainman <alex.brainman@gmail.com>

### CODE REVIEW

File: wgengine/monitor/monitor_windows.go [hunk 1]
```diff
@@ -87,6 +87,7 @@ func (m *winMon) Close() error {
 	m.cancel()
 	m.pollTicker.Stop()
 	windows.SetEvent(m.closeHandle) // wakes up any reader blocked in Receive
+	windows.CloseHandle(m.closeHandle)
```
Comment (bradfitz) [inline wgengine/monitor/monitor_windows.go:90]: Do we need to wait for the goroutine blocked in `WaitMultipleObjects` (or not-yet-blocked in it) to return before closing the handle?

Comment (alexbrainman) [reply wgengine/monitor/monitor_windows.go:90]: We do indeed. I did not consider scenario where m.closeHandle can be used after it was closed. Fixed now.

I did not want to introduce extra complexity. But I don't see other way.

**Comment by alexbrainman** [comment]
@bradfitz do you still want this change?

I think we should submit it. I don't have any alternative version yet.

Alex

**Comment by bradfitz** [comment]
I fixed a typo and updated the commit message to remove inaccurate mentions of `winMin`, which isn't involved here.

**Comment by alexbrainman** [comment]
> I fixed a typo and updated the commit message to remove inaccurate mentions of `winMin`, which isn't involved here.

Sounds good. Thank you.

It was meant to be `winMon.closeHandle`, not `winMin.closeHandle`. But `closeHandle` is good enough too.

Alex

**Pull request merged**